src/youtube_id_search.py

```diff
@@ -70,20 +71,39 @@ def _find_time_string_format(time_str: str) -> str:
 def find_best_matching_youtube_id(db_entry: Dict, search_results: List[dict]) -> str:
     """Function to pick the best match out of the results.
     
-    We accept the ordering provided by the Youtube API and then take the first video
-    with a duration that is close enough to the duration of the song on spotify.
+    We first try to find a video with duration close to the Spotify track (5% threshold).
+    If no exact match is found, we fall back to the best available match within 50% threshold,
+    prioritizing by YouTube's relevance ranking.
     """
+    db_duration = db_entry[COLUMN_TRACK_DURATION]
+    
+    # First pass: try to find exact matches within strict threshold
     for result in search_results:
-        if _is_video_duration_acceptable(
-            db_entry[COLUMN_TRACK_DURATION], result["Duration (s)"]
-        ):
+        if _is_video_duration_acceptable(db_duration, result["Duration (s)"], strict=True):
             return result["ID"]
+    
+    # Second pass: find fallback matches within relaxed threshold
+    fallback_candidates = []
+    for result in search_results:
+        if _is_video_duration_acceptable(db_duration, result["Duration (s)"], strict=False):
+            fallback_candidates.append(result)
+    
+    if fallback_candidates:
+        # Return the first (most relevant) fallback candidate
+        return fallback_candidates[0]["ID"]
```

Comment:
```suggestion
        if _is_video_duration_acceptable(db_duration, result["Duration (s)"], strict=False):
            return result["ID"]
```

um bocadinho mais simples, fica igual ao 1º loop e dá o mesmo resultado